src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -155,3 +266,142 @@ def asset_download(**args):
+                pass
+
+
+def diff_manifest(
+    asset_manager: S3AssetManager,
+    asset_root_manifest: AssetRootManifest,
+    manifest: str,
+    update: bool,
+) -> List[tuple]:
+    """
+    Gets the file paths in specified manifest if the contents of file have changed since its last snapshot.
+    """
+    manifest_dir_name = os.path.basename(manifest)
+    root_path = asset_root_manifest.root_path
+    input_paths: List[Path] = []
+
+    asset_manifest = asset_root_manifest.asset_manifest
+    if asset_manifest is None:
+        raise NonValidInputError("Manifest object not found, please check input manifest. ")
+
+    for base_manifest_path in asset_manifest.paths:
+        if base_manifest_path.path.startswith(manifest_dir_name):
+            # skip the manifest folder, or else every upload will need an update after a previous change
+            continue
+        input_paths.append(Path(root_path, base_manifest_path.path))
+
+    return find_file_with_status(
+        asset_manager=asset_manager,
+        input_paths=input_paths,
+        root_path=root_path,
+        update=update,
+        statuses=[FileStatus.NEW, FileStatus.MODIFIED],
+    )
+
+
+def find_file_with_status(
+    asset_manager: S3AssetManager,
+    input_paths: List[Path],
+    root_path: str,
+    update: bool,
+    statuses: List[FileStatus],
+) -> List[tuple]:
+    """
+    Checks a manifest file, compares it to specified root directory or manifest of files with the local hash cache, and finds files that match the specified statuses.
+    Returns a list of tuples containing the file information, and its corresponding file status.
+    """
+    cache_config = config_file.get_cache_directory()
+
+    with HashCache(cache_config) as hash_cache:
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = {
+                executor.submit(
+                    asset_manager._process_input_path,
+                    path=path,
+                    root_path=root_path,
+                    hash_cache=hash_cache,
+                    update=update,
+                ): path
+                for path in input_paths
+            }
+            status_paths: List[tuple] = []
+            for future in concurrent.futures.as_completed(futures):
+                (file_status, _, manifestPath) = future.result()
+                if file_status in statuses:
+                    status_paths.append((file_status, manifestPath))
+
+            return status_paths
+
+
+def update_manifest(manifest: str, new_or_modified_paths: List[tuple]) -> BaseAssetManifest:
+    """
+    Updates the local manifest file to reflect modified or new files
+    """
+    input_files = glob.glob(os.path.join(manifest, "*_input"))
+
+    if not input_files:
+        raise ValueError(f"No manifest files found in {manifest}")
+    elif len(input_files) >= 2:
+        raise NonValidInputError(f"Does not support multiple manifest_input files in {manifest}. ")
```

Comment:
```suggestion
        raise NonValidInputError(f"Multiple input manifest files are not supported, but found {input_files}.")
```
Same suggestion here for the error message to be consistent